opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -0,0 +1,411 @@
+        let schema_md5 = md5::compute(schema_bytes).0;
+
+        (
+            CentralSchemaEntry {
+                id: schema_id,
+                md5: schema_md5,
+                schema,
+            },
+            field_info,
+        )
+    }
+
+    /// Write row data directly from LogRecord
+    fn write_row_data(&self, log: &LogRecord, field_info: &[FieldInfo]) -> Vec<u8> {
+        let mut buffer = Vec::with_capacity(1024);
+
+        // Sort by order_id to write in schema order
+        let mut sorted_fields = field_info.to_vec();
+        sorted_fields.sort_by_key(|f| f.order_id);
+
+        for field in sorted_fields {
+            match field.name.as_str() {
+                "env_name" => BondWriter::write_string(&mut buffer, "TestEnv"), // TODO - placeholder for actual env name
+                "env_ver" => BondWriter::write_string(&mut buffer, "4.0"), // TODO - placeholder for actual env name
+                "timestamp" | "env_time" => {
+                    let dt = Self::format_timestamp(log.observed_time_unix_nano);
+                    BondWriter::write_string(&mut buffer, &dt);
+                }
+                "env_dt_traceId" => {
+                    if !log.trace_id.is_empty() {
+                        let hex = hex::encode(&log.trace_id);
+                        BondWriter::write_string(&mut buffer, &hex);
+                    }
+                }
+                "env_dt_spanId" => {
+                    if !log.span_id.is_empty() {
+                        let hex = hex::encode(&log.span_id);
+                        BondWriter::write_string(&mut buffer, &hex);
+                    }
+                }
+                "env_dt_traceFlags" => {
+                    if log.flags != 0 {
+                        BondWriter::write_int32(&mut buffer, log.flags as i32);
+                    }
+                }
+                "name" => {
+                    if !log.event_name.is_empty() {
+                        BondWriter::write_string(&mut buffer, &log.event_name);
+                    }
+                }
+                "SeverityNumber" => BondWriter::write_int32(&mut buffer, log.severity_number),
+                "SeverityText" => {
+                    if !log.severity_text.is_empty() {
+                        BondWriter::write_string(&mut buffer, &log.severity_text);
+                    }
+                }
+                "body" => {
+                    // TODO - handle all types of body values
+                    // For now, we only handle string values
+                    if let Some(body) = &log.body {
+                        if let Some(Value::StringValue(s)) = &body.value {
+                            BondWriter::write_string(&mut buffer, s);
+                        }
+                    }
+                }
+                _ => {
+                    // Handle dynamic attributes
+                    if let Some(attr) = log.attributes.iter().find(|a| a.key == field.name) {
```

Comment:
We could do better than scanning all the attributes. We should be able to do all of this in one pass. We could update the `determine_fields` method to also return a vec of bytes which has the bond serialized values.